File: apps/pixelart.py
"""TinyOLED Desktop — Pixel Art Editörü"""
from core.framebuffer import Framebuffer

class PixelArtApp:
    NAME="pixelart"; LABEL="Pixel"; ICON="paint"
    def __init__(self, on_exit):
        self.on_exit=on_exit; self.cx=64; self.cy=32; self.drawing=False
        self.canvas=set()
    def on_up(self):
        if self.drawing: self.cy=max(0,self.cy-1); self.canvas.add((self.cx,self.cy))
        else: self.cy=max(0,self.cy-1)
    def on_down(self):
        if self.drawing: self.cy=min(63,self.cy+1); self.canvas.add((self.cx,self.cy))
        else: self.cy=min(63,self.cy+1)
    def on_sel(self):
        if self.drawing: self.cx=min(127,self.cx+1); self.canvas.add((self.cx,self.cy))
        else: self.drawing=not self.drawing
    def on_long(self):
        if self.drawing: self.drawing=False
        else: self.on_exit()
    def update(self): pass
    def draw(self, fb: Framebuffer):
        for x,y in self.canvas: fb.pixel(x,y)
        # İmleç
        if int(__import__("time").monotonic()*4)%2:
            fb.pixel(self.cx,self.cy)
            fb.pixel(self.cx-1,self.cy); fb.pixel(self.cx+1,self.cy)
            fb.pixel(self.cx,self.cy-1); fb.pixel(self.cx,self.cy+1)
        fb.text("D" if self.drawing else "M",0,0)

File: web-simulator/js/apps/pixel_art.js
import { BrowserApp, loadJSON, saveJSON } from '../browser_app.js';
const KEY='tinyoled.pixelart.v1', W=32, H=11;

export class PixelArtApp extends BrowserApp {
  static NAME='pixel'; static LABEL='Pixel'; static ICON='paint';
  constructor(onExit){super(onExit,{title:'Pixel Art',mode:'WEB'});this.bits=loadJSON(KEY,Array(W*H).fill(0));this.cursor=0;}
  onUp(){this.cursor=(this.cursor-1+W*H)%(W*H);}
  onDown(){this.cursor=(this.cursor+1)%(W*H);}
  onSel(){this.bits[this.cursor]=this.bits[this.cursor]?0:1;saveJSON(KEY,this.bits);this.status='kayit';}
  draw(fb){
    this.drawHeader(fb,'WEB');
    for(let y=0;y<H;y++)for(let x=0;x<W;x++){const i=y*W+x;if(this.bits[i])fb.rect(x*4,21+y*3,4,3,true,true);}
    const cx=(this.cursor%W)*4, cy=21+Math.floor(this.cursor/W)*3; fb.rect(cx,cy,4,3,true,false);
    this.footer(fb,'UP/DN pixel SEL boya');
  }
}
